src/Features/Core/Portable/Completion/Providers/AbstractInternalsVisibleToCompletionProvider.cs

```diff
@@ -0,0 +1,218 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Completion.Providers
+{
+    internal abstract class AbstractInternalsVisibleToCompletionProvider : CommonCompletionProvider
+    {
+        private const string ProjectGuidKey = nameof(ProjectGuidKey);
+
+        protected abstract IImmutableList<SyntaxNode> GetAssemblyScopedAttributeSyntaxNodesOfDocument(SyntaxNode documentRoot);
+        protected abstract SyntaxNode GetConstructorArgumentOfInternalsVisibleToAttribute(SyntaxNode internalsVisibleToAttribute);
+
+        internal override bool IsInsertionTrigger(SourceText text, int insertedCharacterPosition, OptionSet options)
+        {
+            var ch = text[insertedCharacterPosition];
+            return ch == '\"';
```

Comment:
@rchande 
I looked into all the other `IsInsertionTrigger` implementations but I couldn't find anything that fits here. So I decided to come with a first proposal to have a discussion started on how to implement this. The idea is to trigger the provider if there is a double quote somewhere on the same line left of the cursor position.

I thought it might also be useful to check whether the line starts with `[assembly:` but unfortunately `[assembly:` is not required to be at the same line.

```CSharp
internal override bool IsInsertionTrigger(SourceText text, int insertedCharacterPosition, OptionSet options)
{
    var linePosition = text.Lines.GetLinePosition(insertedCharacterPosition);
    var beginOfLine = text.Lines[linePosition.Line].Start;
    var characterPositionInLine = insertedCharacterPosition;
    while (characterPositionInLine >= beginOfLine)
    {
        if (text[characterPositionInLine] == '\"')
        {
            return true;
        }

        characterPositionInLine--;
    }

    return false;
}
```

Does this come close to what you had in mind?